.github/resources/.actions/cli-runner/action.yml

```
@@ -0,0 +1,130 @@
+name: Slack CLI Installation and Command Runner 
+description: Download and cache the Slack CLI and run the input command 
+
+inputs:
+  command:
+    description: "Slack CLI command to run"
+    type: string
+    default: ""
+    required: true
+  verbose:
+    description: "Verbose flag"
+    type: boolean
+    default: false
+    required: false
+  cli_version:
+    description: "CLI Version"
+    type: string
+    default: "latest"
+    required: false
+  app_id: 
+    description: "App ID"
+    type: string
+    default: ""
+    required: false
+
+outputs:
+  success:
+    description: "Whether the command ran successfully"
+    value: ${{ steps.run-slack-cli-command.outputs.success }}
+  exit_code:
+    description: "Exit code"
+    value: ${{ steps.run-slack-cli-command.outputs.exit_code }}
+  command_executed:
+    description: "Command ran"
+    value: ${{ steps.run-slack-cli-command.outputs.command_executed }}
+  stdout:
+    description: "Command output"
+    value: ${{ steps.run-slack-cli-command.outputs.stdout }}
+
+runs:
+  using: composite
+  steps:
+    - name: Cache Slack CLI 
+      id: cache-cli
+      uses: actions/cache@v4
```

```suggestion
      uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
```

📍 suggestion: Let's use a pinned version here to avoid unexpected upstream changes!